docs/rtt/rtt-lua-cookbook.rst

```diff
@@ -0,0 +1,1765 @@
+    sup = d:getPeer("Supervisor")
+
+    sup:exec_file("launch_fsm.lua")
+    sup:configure()
+    cmd = rttlib.port_clone_conn(sup:getPort("events"))
+
+..
+
+Run it. cmd is an inverse (output) port which is connected to the incoming (from POV of the fsm) 'events' port of the fsm, so by writing to it we can send events:
+
+.. code-block:: bash
+
+    $ rosrun ocl rttlua-gnulinux -i deploy.lua
+    OROCOS RTTLua 1.0-beta3 / Lua 5.1.4 (gnulinux)
+    INFO: created undeclared connector root.initial
+    > sup:start()
+    > in ping entry
+
+    > cmd:write("e_pong")
+    > in pong entry
+
+    > cmd:write("e_ping")
+    > in ping entry
+
+    > cmd:write("e_pong")
+    > in pong entry
+
+..
+
+**Option B: Running the rFSM example with an Orocos deployment script**
+
+**deploy.ops**
+
+::
+
+    import("ocl")
+    loadComponent("Supervisor", "OCL::LuaComponent")
+    Supervisor.exec_file("launch_fsm.lua")
+    Supervisor.configure
+
+..
+
+After starting the supervisor we 'leave' it, so we can write to the 'events' ports:
+
+.. code-block:: bash
+
+    $ rosrun ocl deployer-gnulinux -s deploy.ops
+    INFO: created undeclared connector root.initial
+    Switched to : Deployer
+
+        This console reader allows you to browse and manipulate TaskContexts.
+        You can type in an operation, expression, create or change variables.
+        (type 'help' for instructions and 'ls' for context info)
+
+            TAB completion and HISTORY is available ('bash' like)
+
+    Deployer [S]> cd Supervisor
+
+    TaskBrowser connects to all data ports of Supervisor
+    Switched to : Supervisor
+    Supervisor [S]> start
+    = true
+
+    Supervisor [R]> in ping entry
+
+    Supervisor [R]> leave
+    Watching Supervisor [R]> events.write ("e_pong")
+    = (void)
+
+    Watching Supervisor [R]> in pong entry
+
+    Watching Supervisor [R]> events.write ("e_ping")
+    = (void)
+
+    Watching Supervisor [R]> in ping entry
+
+    Watching Supervisor [R]>
+
+..
+
+Running rFSM in a Service
+-------------------------
+
+This is basically the same as executing a function periodally in a service (see the Service example above). There is a convenience function service_launch_rfsm in rfsm_rtt.lua to make this easier.
+
+The steps are:
+
+    - create LuaService in Component in question
+    - prepare Lua environment, i.e. call exec_string or exec_file to add functions.
+    - launch the fsm with the following call in your deployment script:
+
+.. code-block:: lua
+
+    require "rfsm_rtt"
+
+    -- get reference to exec_str operation
+    fsmfile = "fsm.lua"
+    execstr_op = comp:provides("Lua"):getOperation("exec_str")
+    rfsm_rtt.service_launch_rfsm(fsmfile, execstr_op, true)
+
+..
+
+The last line means the following: launch fsm in <fsmfile> in service identified by execstr_op, true: create an execution engine hook so that the rfsm.step is called at the component frequency. (See the generated rfsm_rtt API docs).
```

Comment:
```suggestion
The last line means the following: launch fsm in ``<fsmfile>`` in service identified by ``execstr_op``, true: create an execution engine hook so that the ``rfsm.step`` is called at the component frequency. (See the generated ``rfsm_rtt`` API docs).
```